src/extension/src/TelemetryWriter.py

```
@@ -13,75 +13,107 @@
+        if len(json.dumps(event)) > Constants.TELEMETRY_EVENT_SIZE_LIMIT_IN_CHARACTERS:
+            print("Cannot send data to telemetry as it exceeded the acceptable data size. [Data not sent={0}]".format(json.dumps(message)))
+        else:
+            self.write_using_temp_file(self.events_folder_path, event)
+
+    def __delete_older_events(self):
+        """ Delete older events until the atleast one new event file can be added as per the size restrictions """
+        try:
+            if os.path.getsize(self.events_folder_path) < Constants.TELEMETRY_DIR_SIZE_LIMIT_IN_CHARACTERS - Constants.TELEMETRY_EVENT_FILE_SIZE_LIMIT_IN_CHARACTERS:
+                print("Not deleting any existing event files as the event directory does not exceed max limit. At least one new event file can be added.")
+                return
+
+            print("Events directory size exceeds maximum limit. Deleting older event files until at least one new event file can be added.")
+            event_files = [os.path.join(self.events_folder_path, event_file) for event_file in os.listdir(self.events_folder_path) if (event_file.lower().endswith(".json"))]
+            event_files.sort(key=os.path.getmtime, reverse=True)
+
+            for event_file in event_files:
+                try:
+                    if os.path.getsize(self.events_folder_path) < Constants.TELEMETRY_DIR_SIZE_LIMIT_IN_CHARACTERS - Constants.TELEMETRY_EVENT_FILE_SIZE_LIMIT_IN_CHARACTERS:
+                        print("Not deleting any more event files as the event directory has sufficient space to add at least one new event file")
+                        break
+
+                    if os.path.exists(event_file):
+                        os.remove(event_file)
+                        print("Deleted [File={0}]".format(repr(event_file)))
+                except Exception as e:
+                    print("Error deleting event file. [File={0}] [Exception={1}]".format(repr(event_file), repr(e)))
```

Since this is swallowing exceptions, what's the recovery/continuation strategy if there are rampant failures (like all event files can't be deleted)? At what point is it a terminal exception?

---

Changed to throw a terminal exception is no files were deleted or the directory size exceeds limit even after deletion attempts